src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -1050,6 +1050,7 @@ private void LogFailedProbe(SiloAddress silo, int pingNumber, Exception failureR
 
                 if (logger.IsVerbose) logger.Verbose("-Going to DeclareDead silo {0} in the table. About to write entry {1}.", entry.SiloAddress.ToLongString(), entry.ToFullString());
                 entry.Status = SiloStatus.Dead;
+                RuntimeClient.Current.BreakOutstandingMessagesToDeadSilo(entry.SiloAddress);
```

Comment:
This is definitely, absolutely, out of place.
1) It is too early, the silo has not been killed yet.
2) this is an internal impl. of membership. We should not be changing that. There is a well defined "public" API notification that membership exposes to listeners (it is "public" in the sense of that component to other component's. It is still runtime internal). So the needed components will receive a notification about this silo dead at the right time. That is what the Catalog will get.

Just remove that line.